### ISSUE TITLE
 training imagenet on own data segmentation fault 

### ISSUE THREAD

**Issue by obrianjunior** [BVLC/caffe#2672]
Hi everyone,  i'm trying  to  train  imagenet on my onw data as describe in the tutorial.
I've followed all the steps and sadly when i  try to  train the network i got this error :

I0701 14:00:50.278710 15058 net.cpp:247] Network initialization done.
I0701 14:00:50.278717 15058 net.cpp:248] Memory required for data: 68946528
I0701 14:00:50.278825 15058 solver.cpp:42] Solver scaffolding done.
I0701 14:00:50.278868 15058 solver.cpp:250] Solving CaffeNet
I0701 14:00:50.278879 15058 solver.cpp:251] Learning Rate Policy: step
I0701 14:00:50.376868 15058 solver.cpp:294] Iteration 0, Testing net (#0)
*** Aborted at 1435759251 (unix time) try "date -d @1435759251" if you are using GNU date ***
PC: @     0x7ffdd411fd24 caffe::SoftmaxWithLossLayer<>::Forward_cpu()
*** SIGSEGV (@0xf372c5c0) received by PID 15058 (TID 0x7ffdd4678780) from PID 18446744073498969536; stack trace: ***
    @     0x7ffdd34c3d40 (unknown)
    @     0x7ffdd411fd24 caffe::SoftmaxWithLossLayer<>::Forward_cpu()
    @     0x7ffdd40a9032 caffe::Net<>::ForwardFromTo()
    @     0x7ffdd40a9327 caffe::Net<>::ForwardPrefilled()
    @     0x7ffdd4093e6f caffe::Solver<>::Test()
    @     0x7ffdd4094826 caffe::Solver<>::TestAll()
    @     0x7ffdd409d9cd caffe::Solver<>::Step()
    @     0x7ffdd409e39f caffe::Solver<>::Solve()
    @           0x407336 train()
    @           0x405761 main
    @     0x7ffdd34aeec5 (unknown)
    @           0x405d0d (unknown)
Segmentation fault (core dumped)

I've been looking through this forum and others , but it seems no one has allready got this error.    Also i've tried with both  lmdb  input and  raw image input i got the same error.
Can someone help me please ?

**Comment by ronghanghu** [comment]
I guess a likely cause for your crash is that your labels wrong (e.g. you have 200 classes while you have some label == 200 or 201 or -1), causing array subscript out of bounds in SoftmaxWithLossLayer and consequently segmentation fault.

This tracker is for issues and bugs with Caffe. Please ask usage questions on the [caffe-users](https://groups.google.com/forum/#!forum/caffe-users) list. Thanks!

**Comment by obrianjunior** [comment]
Thanks @ronghanghu  ! indeed my labels are very big int ( 1e-9 like ) while i have always 2875 classes.  I did not knew caffe requires the labels to be explicitly coded from 0  to  numberOfClasses-1.

I try fixing this, and come back to the forum to post the result.  Thanks !

**Comment by obrianjunior** [comment]
It's Works when recoding the labels from 0 to numberOfclasses-1.
(y) .

**Comment by longjon** [comment]
Closing as there does not seem to be a bug here. In the future, you may want to try running in `DEBUG` mode, which explicitly checks for things like this.

**Comment by AmlanAlok** [comment]
@obrianjunior Hi

I am facing the same problem. Can you tell what do you mean by " I did not knew caffe requires the labels to be explicitly coded from 0 to numberOfClasses-1." ?
